zeebe/atomix/cluster/src/test/java/io/atomix/raft/FaultyFlusherConfigurator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.atomix.raft;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.raft.RaftRule.Configurator;
+import io.atomix.raft.RaftServer.Builder;
+import io.atomix.raft.partition.RaftElectionConfig;
+import io.atomix.raft.storage.RaftStorage;
+import io.atomix.raft.storage.log.RaftLogFlusher;
+import io.camunda.zeebe.journal.CheckedJournalException.FlushException;
+import io.camunda.zeebe.journal.Journal;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Set the first faultyFlusherNumber nodes with a faulty flusher, when the supplier returns true */
+public record FaultyFlusherConfigurator(
+    int faultyFlusherNumber,
+    Supplier<Boolean> faultyWhen,
+    Runnable notifyFaultyFlush,
+    boolean leaderFaulty,
+    boolean withDataLoss)
+    implements Configurator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(FaultyFlusherConfigurator.class);
+
+  private RaftLogFlusher.Factory faultyFlusher(
+      final Supplier<Boolean> faultyWhen, final Runnable notifyFaultyFlush) {
+    return (ignored) ->
+        new RaftLogFlusher() {
+          @Override
+          public void flush(final Journal journal) throws FlushException {
+            if (faultyWhen.get()) {
+              notifyFaultyFlush.run();
+              if (withDataLoss) {
+                journal.deleteAfter(journal.getLastIndex() - 1);
```

Comment:
Do you think this addition is worth?

Reply:
Hm, I'm not sure what's the additional test coverage with this case. We already have data loss tests. So here specifically it would be testing data loss on flush error, right? I don't think it's harmful, but I'm not seeing the big advantage. Can you elaborate on it?